### ISSUE TITLE
 Kuun-Lan's engines flashing when firing

### ISSUE THREAD

**Issue by DesolationStone** [ncblakely/CataclysmEngineTweaks#17]
As title, I've found that the engines of the Kuun-Lan's starts to flash when the ship it's firing with her point defense turrets. I don't know if this is a bug present in the vanilla game, but I report it anyway.

**Comment by Talthos** [comment]
Pretty sure this happens in the vanilla game, yes. It also occurs with most capital ships in the game, if I recall correctly.

**Comment by SuperSajuuk** [comment]
Yeah, that happens in HW1 Classic as well iirc. As much as I'd like this to be fixed in CET for Cataclysm, its inconsequential and has no bearing on anything, other than being funky I guess.

**Comment by rbrune** [comment]
I wonder if it is the same issue that I fixed in the HW1 engine. The mothership engine kept flickering in the intro sequence. Turns out there is a check to see if velocity == 0 to turn off the engine effect and as soon as velocity != 0 the effect is started again. Due to physics engine floating point precision the velocity is switching back and forth between very small values and 0. Introducing a small epsilon helped fix that.

https://gitlab.com/gardens-of-kadesh/gardens-of-kadesh/-/commit/b0a4daa19368afc0e128e4aab57f07f69d8b0c90

I just noticed my fix is only for NIS playback - but I assume there is similar code somewhere else.
A quick search through the code later: there is a macro for setting an object to moving which then causes the engine effect to be drawn:

`SET_MOVING_LINEARLY(obj->posinfo.isMoving);`

It is used in multiple places in Physics.c and the test for if something is moving looks like this:

`if (Real32ToUdword(obj->posinfo.velocity.x) | Real32ToUdword(obj->posinfo.velocity.y) | Real32ToUdword(obj->posinfo.velocity.z))`

So as above velocity needs to be perfectly 0 for it to turn off the engine effect.
